### PR TITLE
Fix BMM ack issues

### LINKED ISSUE #70: M7 messages always ingored
M7 messages in a coinbase tx are always ignored, due to a missing case here
https://github.com/LayerTwo-Labs/bip300301_enforcer/blob/master/src/messages.rs#L208

### PR DESCRIPTION
**Pull request by Ash-L2L**
Fixes #70, #71, #72

### CODE REVIEW

**Review by torkelrogstad — APPROVED** [review]

File: src/wallet/mod.rs [hunk 1]
```diff
@@ -49,13 +49,36 @@ use crate::{
...
     types::{Ctip, SidechainAck, SidechainNumber, SidechainProposal},
     validator::Validator,
 };
 use error::WalletError;
 
 pub mod error;
 
+#[derive(Debug)]
+pub struct Deposit {
+    pub sidechain_number: u8,
```
Comment (torkelrogstad) [inline src/wallet/mod.rs:62]: `SidechainNumber`?

File: src/wallet/mod.rs [hunk 2]
```diff
@@ -49,13 +49,36 @@ use crate::{
...
+    pub address: Vec<u8>,
+    pub amount: u64,
```
Comment (torkelrogstad) [inline src/wallet/mod.rs:64]: `bitcoin::amount::Amount`?

**Review by torkelrogstad — APPROVED** [review]
Just minor stuff, not important. 

For next time: would be great if code moves were in separate commits from the actual changes. Makes it a lot easier to review the things that _do_ change

File: src/wallet/mod.rs (hunk 2 above)
Comment (nchashch) [reply src/wallet/mod.rs:64]: I think u64 is better for internal stuff, `Amount` is useful for displaying and parsing Bitcoin amounts but there isn't much benefit to using it internally, it just adds more boilerplate.

Also it isn't the kind of situation where you are likely to make a type error -- you probably won't try to compute a sum of `amount` and `block_height`.

File: src/wallet/mod.rs (hunk 1 above)
Comment (nchashch) [reply src/wallet/mod.rs:62]: I don't think the type wrapper for `sidechain_number` makes sense, it is already clear what it is just by the field name, adding the type wrapper just adds boilerplate without any clear benefit.

**Review by nchashch — APPROVED** [review]

**Pull request merged**